### PR TITLE
Fix DHO state overwritten at `LoadComplete` if it is already changed

### LINKED ISSUE #14126: Starting to spectate an osu!mania match from mid-gameplay causes performance issues
"Refreshes" count goes from a normal value (~1.5k with two players when spectating from the beginning of the track) to 20-40k. Likely lifetime related, unsure if osu!mania specific.

https://user-images.githubusercontent.com/191335/128123840-3bb0c3bf-0842-47bb-952b-edcebcd2cb6c.mp4

### PR DESCRIPTION
**Pull request by ekrctb**
- Fixes #14126 (see my comment)

Note: frame-unstable seek (used by the spectator) will load many hit objects in one frame, causes a massive lag. But that is separate from the linked issue, as the linked issue is persistent after seek, not just one frame.

### CODE REVIEW

File: osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs [hunk 1]
```diff
@@ -190,7 +190,9 @@ protected override void LoadComplete()
             comboIndexBindable.BindValueChanged(_ => UpdateComboColour());
             comboIndexWithOffsetsBindable.BindValueChanged(_ => UpdateComboColour(), true);
 
-            updateState(ArmedState.Idle, true);
+            // If the state is changed, transforms are already initialized.
+            if (state.Value == ArmedState.Idle)
```
Comment (bdach) [inline osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs:194]: Adding this guard is the same as specifying `force: false` in `updateState()`, is it not? Which makes me wonder why the true was there in the first place and whether doing this won't break something else.

Comment (ekrctb) [reply osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs:194]: It is the opposite. `force: false` will apply transforms when the current state is *not* `Idle`.

Comment (bdach) [reply osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs:194]: ...which is different from the guard how? I'm confused. The guard is checking for idle, and not applying transforms forcefully if idle...?

Comment (bdach) [reply osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs:194]: Never mind, you're correct.

**Comment by ekrctb** [comment]
The test failure of `TestSliderAnimationDisable` is relevant.
This breaks slider in editor. Not sure why, yet.

File: osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs (hunk 1 above)
Comment (frenzibyte) [reply osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs:194]: It reads backwards though, the rare case is when state changes before load, but most of the time it wouldn't. An early-return would make it read better:
```csharp
// The state has already been changed, no need to initialize transforms.
if (state.Value != ArmedState.Idle)
    return;

updateState(ArmedState.Idle, true);
```

**Comment by ekrctb** [comment]
Although I didn't confirm exactly why the slider was broken, that is probably because the state change is before the nested drawables are added. To not break existing code such as this, I changed the way to solve the original issue.
Now, it is just using the current state instead of `ArmedState.Idle`.

File: osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs (hunk 1 above)
Comment (ekrctb) [reply osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs]: This change is obsolete.

**Review by smoogipoo — APPROVED** [review]
lgtm

**Pull request merged**